identity-apps-core/apps/recovery-portal/src/main/webapp/password-recovery-otp.jsp

```diff
@@ -0,0 +1,315 @@
+        }
+        request.getRequestDispatcher("sms-otp.jsp").forward(request, response);
+    } else if (RecoveryStage.CONFIRM.equalsValue(recoveryStage)) {
+        String flowConfirmationCode = request.getParameter("flowConfirmationCode"); 
+        String OTPcode = request.getParameter("OTPcode");
+        try {
+            Map<String, String> requestHeaders = new HashedMap();
+            if (request.getParameter("g-recaptcha-response") != null) {
+                requestHeaders.put("g-recaptcha-response", request.getParameter("g-recaptcha-response"));
+            }
+            ConfirmRequest confirmRequest = new ConfirmRequest();
+            /** For local notification channels flowConfirmationCode is used as confirmation code. */
+            confirmRequest.setConfirmationCode(flowConfirmationCode);
+            confirmRequest.setOtp(OTPcode);
+            ConfirmResponse confirmResponse = 
+                recoveryApiV2.confirmPasswordRecovery(confirmRequest, tenantDomain, requestHeaders);
+            request.setAttribute("resetCode", confirmResponse.getResetCode());
+        } catch (ApiException e) {
+            if (!StringUtils.isBlank(username)) {
+                request.setAttribute("username", username);
+            }
+            /** Status code 406 is used for invalid/expired channel id/recovery code. Other error are considered
+            unexpected and redirected to the error page. */
+            if (e.getCode() != 406) {
+                IdentityManagementEndpointUtil.addErrorInformation(request, e);
+                request.getRequestDispatcher("error.jsp").forward(request, response);
+                return;
+            }
+            request.setAttribute("isAuthFailure","true");
+            request.setAttribute("authFailureMsg", "authentication.fail.message");
+            request.setAttribute("resendCode", request.getParameter("resendCode"));
+            request.setAttribute("flowConfirmationCode", request.getParameter("flowConfirmationCode"));
+            request.getRequestDispatcher("sms-otp.jsp").forward(request, response);
+            return;
+        }
+        request.setAttribute("spId", request.getParameter("spId"));
+        request.getRequestDispatcher("password-reset.jsp").forward(request, response);
+    } else if (RecoveryStage.RESET.equalsValue(recoveryStage)) {
+        request.setAttribute("useRecoveryV2API", "true");
+        request.setAttribute("spId", request.getParameter("spId"));
+        request.getRequestDispatcher("password-reset-complete.jsp").forward(request, response);
+    } else {
+        request.setAttribute("errorMsg", "Invalid password recovery stage.");
+        if (!StringUtils.isBlank(username)) {
+            request.setAttribute("username", username);
+        }
+        request.getRequestDispatcher("error.jsp").forward(request, response);
+    }
+%>
+
+<!doctype html>
+<html lang="en-US">
+<head>
+</head>
+<body>
+</body>
+</html>
```

Comment:
We don't need this blank html template here, do we?

Reply:
fixed with https://github.com/wso2/identity-apps/pull/5817/commits/c4711e788c5f3c08e1e2906f20e2217b381f34e9